zerver/lib/events.py

```diff
@@ -392,8 +392,6 @@ def apply_event(state: Dict[str, Any],
                 if 'gravatar.com' in person['avatar_url']:
                     person['avatar_url'] = None
             person['is_active'] = True
-            if not person['is_bot']:
-                person['profile_data'] = {}
```

Comment:
I have removed this as `profile_data` is already passed as `{}` (and of course with original data in reactivation), so we do not require this.